src/thor/costmatrix.cc

```diff
@@ -821,22 +821,31 @@ void CostMatrix::CheckForwardConnections(const uint32_t source,
 
       // if source percent along edge is larger than target percent along,
       // can't connect on this edge
-      if (find_correlated_edge(options.sources(source), fwd_pred.edgeid()).percent_along() >
-          find_correlated_edge(options.targets(target), fwd_pred.edgeid()).percent_along()) {
+      const auto source_edge = find_correlated_edge(options.sources(source), fwd_pred.edgeid());
+      const auto target_edge = find_correlated_edge(options.targets(target), fwd_pred.edgeid());
+      if (source_edge.percent_along() > target_edge.percent_along()) {
         continue;
       }
 
-      // remember: transition_cost is abused in SetSources/Targets: cost is secs, secs is length
-      float s =
-          std::abs(fwd_pred.cost().secs + rev_label.cost().secs - rev_label.transition_cost().cost);
+      // initial labels' transition cost is the full edge cost
+      Cost partial_cost = fwd_pred.cost() + rev_label.cost() -
+                          ((fwd_pred.transition_cost() + rev_label.transition_cost()) * 0.5);
```

Comment:
is this correct math? not sure it was correct math before, but this is definitely different. would be good to flag these things in the PR and explain the diff. took me quite a while to re-understand where all this is coming from :)